### PR TITLE
add doxygen for g2 lib on a, c, d, e named code files

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
Linked on issue #59
The doxygen has been tested and valided on 
https://hang-lei-noaa.github.io/NCEPLIBS/NCEPLIBS-g2/html/files.html

### CODE REVIEW

File: src/addlocal.f [hunk 1]
```diff
@@ -1,45 +1,32 @@
+!>    @file
+!>    @brief This subroutine adds a Local Use Section (Section 2) to
+!>    a GRIB2 message.
+!>    @author Gilbert ORG: W/NP11 @date 2000-05-01
+!>
```
Comment (edwardhartnett) [inline src/addlocal.f:5]: Need a blank line between file and subroutine blocks.

File: src/cmplxpack.f [hunk 2]
```diff
@@ -1,59 +1,31 @@
+!>    @file
+!>    @brief This subroutine packs up a data field using a complex
+!>    packing algorithm as defined in the GRIB2 documention.
+!>    @author Gilbert ORG: W/NP11 @date 2004-08-27
+!>
```
Comment (edwardhartnett) [inline src/cmplxpack.f:5]: blank line here to separate docblocks.

File: src/cmplxpack.f [hunk 3]
```diff
@@ -1,59 +1,31 @@
...
+!>    @brief This subroutine packs up a data field using a complex
+!>    packing algorithm as defined in the GRIB2 documention.
+!>    @author Gilbert ORG: W/NP11 @date 2004-08-27
+!>     
+!>    This subroutine supports GRIB2 complex packing templates with or without
+!>    spatial differences (i.e. DRTs 5.2 and 5.3).
+!>    It also fills in GRIB2 Data Representation Template 5.2 or 5.3 
+!>    with the appropriate values.
+!>    
+!>    @param[in] fld() Contains the data values to pack
```
Comment (edwardhartnett) [inline src/cmplxpack.f]: Don't use parens when listing arrays. So not fld(), instead just fld.

Only use parens for function/subroutine names.

File: src/compack.f [hunk 4]
```diff
@@ -1,60 +1,33 @@
+!>    @file
+!>    @brief This subroutine packs up a data field using a complex
+!>    packing algorithm as defined in the GRIB2 documention.
+!>    @author Gilbert ORG: W/NP11 @date 2000-06-21
+!>
```
Comment (edwardhartnett) [inline src/compack.f:4]: Blank line...

File: src/compack.f [hunk 5]
```diff
@@ -1,60 +1,33 @@
...
+!>    packing algorithm as defined in the GRIB2 documention.
+!>    @author Gilbert ORG: W/NP11 @date 2000-06-21
+!>     
+!>    It supports GRIB2 complex packing templates with or without
+!>    spatial differences (i.e. DRTs 5.2 and 5.3).
+!>    It also fills in GRIB2 Data Representation Template 5.2 or 5.3 
+!>    with the appropriate values.
+!>     
+!>    @param[in] fld Contains the data values to pack
+!>    @param[in] ndpts The number of data values in array fld()
```
Comment (edwardhartnett) [inline src/compack.f]: Remove parens from fld since it's not a field.

File: src/compack.f [hunk 6]
```diff
@@ -1,60 +1,33 @@
+!>    @file
+!>    @brief This subroutine packs up a data field using a complex
+!>    packing algorithm as defined in the GRIB2 documention.
+!>    @author Gilbert ORG: W/NP11 @date 2000-06-21
+!>     
+!>    It supports GRIB2 complex packing templates with or without
```
Comment (edwardhartnett) [inline src/compack.f]: OK, the information in your brief tag needs to be in two places, for the file and for the subroutine. Look at some of the files I did for examples of how to do this.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]
A few too many files here to review. Let's just do one at a time until we get all the comments out of the way.

File: src/enc_png.c [hunk 7]
```diff
@@ -85,27 +100,35 @@ int SUB_NAME(char *data,g2int *width,g2int *height,g2int *nbits,char *pngbuf)
        return (-2);
     }
 
-/*     Set Error callback   */
+/*
```
Comment (edwardhartnett) [inline src/enc_png.c:104]: You shouldn't have to do anything to the comments in the code itself when doxygenating the file. They can just be left alone.

**Comment by edwardhartnett** [comment]
Actually this has nothing to do with issue #59, it is part of #55.

**Comment by Hang-Lei-NOAA** [comment]
I do add a 5 blank position line to separate these.
will remove parens
Will make the brief more brief
Editing the in-code comments is to make it more uniformed looking in
code, not for doxygen.

**Comment by Hang-Lei-NOAA** [comment]
I will open a new pull request using the way George suggested.

**Pull request closed without merging**